src/d3a/d3a_core/sim_results/area_statistics.py

```diff
@@ -243,13 +291,26 @@ def _accumulate_area_trades(area, parent, accumulated_trades, past_market_types)
         for market in area_markets:
             for trade in market.trades:
                 if area_name_from_area_or_iaa_name(trade.seller) == \
-                        area.name and trade.buyer not in child_names:
-                    accumulated_trades[area.name]["consumedFromExternal"] += trade.offer.energy
-                    accumulated_trades[area.name]["spentToExternal"] += trade.offer.price
+                        area.name and area_name_from_area_or_iaa_name(trade.buyer) in child_names:
+                    accumulated_trades[area.name]["consumedFromExternal"] = \
+                        substract_or_create_key(accumulated_trades[area.name]
+                                                ["consumedFromExternal"],
+                                                area_name_from_area_or_iaa_name(trade.buyer),
+                                                trade.offer.energy)
+                    accumulated_trades[area.name]["spentToExternal"] = \
```

Comment:
Isn't spentToExternal supposed to be always positive? I am not sure on the sign convention on prices, but I have the impression, since both spentToExternal and earnedFromExternal used to only increment, it is probably without a sign convention (always positive values). If this is correct, the subtract_or_create_key call should be replaced with an add_or_create_key.